### ISSUE TITLE
[Ray-tune] Error using Ray tune with Huggingface's Trainer for hyperparameter_search

### ISSUE THREAD

**Issue by phosseini** [ray-project/ray#23981]
### What happened + What you expected to happen

I'm using Ray with Huggingface `Trainer` for hyperparameter tuning.  I'm running my code in a Google Colab Pro+ environment on 1 GPU and 1 CPU. I've changed nothing in my code, but today after installing everything the same way, I'm getting the following error that looks like is coming from Ray:

```
Some weights of BertForMultipleChoice were not initialized from the model checkpoint at phosseini/glucose-bert-large and are newly initialized: ['classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Traceback (most recent call last):
  File "hp_tuning_copa.py", line 171, in <module>
    log_to_file=True)
  File "/usr/local/lib/python3.7/dist-packages/transformers/trainer.py", line 1913, in hyperparameter_search
    best_run = backend_dict[backend](self, n_trials, direction, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/transformers/integrations.py", line 297, in run_hp_search_ray
    **kwargs,
  File "/usr/local/lib/python3.7/dist-packages/ray/tune/tune.py", line 348, in run
    _ray_auto_init()
  File "/usr/local/lib/python3.7/dist-packages/ray/tune/tune.py", line 841, in _ray_auto_init
    ray.init()
  File "/usr/local/lib/python3.7/dist-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/ray/worker.py", line 1023, in init
    head=True, shutdown_at_exit=False, spawn_reaper=True, ray_params=ray_params
  File "/usr/local/lib/python3.7/dist-packages/ray/node.py", line 301, in __init__
    self.start_ray_processes()
  File "/usr/local/lib/python3.7/dist-packages/ray/node.py", line 1130, in start_ray_processes
    resource_spec = self.get_resource_spec()
  File "/usr/local/lib/python3.7/dist-packages/ray/node.py", line 481, in get_resource_spec
    ).resolve(is_head=self.head, node_ip_address=self.node_ip_address)
  File "/usr/local/lib/python3.7/dist-packages/ray/_private/resource_spec.py", line 197, in resolve
    system_memory = ray._private.utils.get_system_memory()
  File "/usr/local/lib/python3.7/dist-packages/ray/_private/utils.py", line 436, in get_system_memory
    docker_limit = int(f.read())
ValueError: invalid literal for int() with base 10: 'max\n'
```
This is also the detailed info on my memory and computation (for tuning, I'm only using 1 GPU and 1 CPU):
```
Mon Apr 18 20:10:18 2022       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 460.32.03    Driver Version: 460.32.03    CUDA Version: 11.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000000:00:04.0 Off |                    0 |
| N/A   36C    P0    25W / 300W |      0MiB / 16160MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU   GI   CI        PID   Type   Process name                  GPU Memory |
|        ID   ID                                                   Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
Your runtime has 54.8 gigabytes of available RAM

You are using a high-RAM runtime!
```

I expect the tuning/running to be completed without any error, as it used to.

### Versions / Dependencies

- Transformers 4.18.0
- Ray 1.12.0 (I installed `! pip install "ray[tune]"`)

### Reproduction script

The fully reproducible script is not available since I have some non-public data. But This is how I'm running my code:

[https://github.com/phosseini/causal-reasoning/blob/main/fine_tuning_copa.py](https://github.com/phosseini/causal-reasoning/blob/main/fine_tuning_copa.py)

### Issue Severity

High: It blocks me from completing my task.

**Comment by amogkam** [comment]
Hey @phosseini yes this is a known issue with Ray 1.12.0 on Google Colab. Please see https://github.com/ray-project/ray/issues/23951 for more information and workarounds!

**Comment by JessicaLopezEspejel** [comment]
Hello, 

[phosseini](https://github.com/phosseini)  did you achieve to resolve the issue? I tried the workarounds suggest by @amogkam , but I have the following error: 

```
AttributeError: module 'pickle' has no attribute 'PickleBuffer'

```
![ray_error](https://user-images.githubusercontent.com/53188008/164421186-365c14d0-a8f7-452f-95ce-ced284bec2a9.PNG)

It looks like the problem is in python 3.7, however when I install python 3.8 in my colab:


```
!wget -O mini.sh https://repo.anaconda.com/miniconda/Miniconda3-py38_4.8.2-Linux-x86_64.sh
!chmod +x mini.sh
!bash ./mini.sh -b -f -p /usr/local
!conda install -q -y jupyter
!conda install -q -y google-colab -c conda-forge
!python -m ipykernel install --name "py38" --user
```
It does not recognize `transformers` , that I already install:


![python3 8_error](https://user-images.githubusercontent.com/53188008/164421737-e96d9e8b-bb67-49d7-b52a-3423f0c85494.PNG)

Thank you so much.

**Comment by JessicaLopezEspejel** [comment]
Hello @phosseini, I am using  6.0.1 version of `pyarrow`, which it is installed by default when I install datasets '!pip install datasets':
![pyarrow](https://user-images.githubusercontent.com/53188008/164738074-a91fdfa8-06e6-4c0a-ae25-3f8c5f9661dc.PNG)

I read in this [link](https://stackoverflow.com/questions/67798070/raytune-is-throwing-error-module-pickle-has-no-attribute-picklebuffer-whe) that it's because of the python version. It says, that instead to use python 3.7, the problem is resolved if we use python 3.8. I tried to install [python 3.8 in Google Colab](https://stackoverflow.com/questions/60775160/install-python-3-8-kernel-in-google-colaboratory), as many people described it. Later I executed `!pip install transformers`. However, if I make work python 3.8,  `transformers` library does work correctly. 

![error_transformers](https://user-images.githubusercontent.com/53188008/164741053-bec27310-8abc-4876-95d6-826c88924d03.PNG)

Do you have any suggestion?

Thank you.
